### DISCUSSION TITLE
Handlebars value reference for '0' issue

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3189] [Help]
Hello,

After upgrade to the latest(Oct 0.9.173), some of my apps got errors, and I find out that Handlebars works differently in some case.

1. The custom query just didn't support handlebar anymore? As some of my query has handlebar function like {{subtract}}, now all just didn't work. It's not an issue as I can calculate outside the query.
2. I have a custom SQL params, before upgrade I can pass a params by `{{#if (lte Form_Customers.Fields.PageNo 1)}}0{{else}}{{times (subtract Form_Customers.Fields.PageNo 1) Form_Customers.Fields.PageSize}}{{/if}}`, but now the same code it always got error. I have bind the same code to a paragraph an I can see the value is correct, but I check the Budibase backend logs and found the query `select .... invalid binding`, it seems handlebar didn't passed the correct params value to query. Luckly, BB can support Javascript now, then I use below javascript and it works just fine. 

```
if ($("Form_Customers.Fields.PageNo") > 1)
	return (parseInt($("Form_Customers.Fields.PageNo"))-1)*parseInt($("Form_Customers.Fields.PageSize"))
else 
	return 0
```

**Comment by samg666** [comment]
I got more errors after upgrade, most are about `0` value references. Before upgrade, for a number field, if I input `0`, then it will pass `0` to my query. But after upgrade, even I have inputed `0`, it still  passed empty `''` to my query, then MySQL raised error `Incorrect integer value: '' for column 'Amount' at row 1`. Although I find a work around, put `{{#if}}` statement to make it work.

Before upgrade
`{{Form1.Fields.Amount}}`

After upgrade, workaround: I have to use this statement to make sure Budibase will pass `0` as custom query params.
`{{#if (lte Form1.Fields.Amount 0)}}0{{else}}{{Form1.Fields.Amount}}{{/if}}`

**Comment by mjashanks** [comment]
> 1. The custom query just didn't support handlebar anymore?

That's right, we had to remove it as it was causing an issue. We decided that the value could just be calculated from the design, as you suggest. Apologies for not communicating this.

> 2. I have a custom SQL params,....

I tried your exact scenario - I copy and pasted your binding, and it worked for me. first, I added a paragraph element to the screen, and tried your binding - just to debug its result. That might help you - although it looks like you have solved it already with javascript.

 So - I'm not sure what the issue is here - it seems to be working as expected for me

> ...  most are about 0 value references ...

This is a pretty serious issue, which I've logged here: https://github.com/Budibase/budibase/issues/3195

Thank you for reporting, we'll get it fixed asap
